manifests/chain.pp

```diff
@@ -1,23 +1,43 @@
 # defined resource which creates all rules for one chain
-# @param policy Set the default policy for a CHAIN
+# @param policy Set the default policy for CHAIN (works only for builtin chains)
 # @param disable_conntrack Disable/Enable usage of conntrack
-# @param chain Name of the chain that should be managed
 # @param log_dropped_packets Enable/Disable logging of packets to the kernel log, if no explicit chain matched
+# @param chain Name of the chain that should be managed
+# @param table Select the target table (filter/raw/mangle/nat)
 define ferm::chain (
-  Ferm::Policies $policy,
   Boolean $disable_conntrack,
   Boolean $log_dropped_packets,
   String[1] $chain = $name,
+  Optional[Ferm::Policies] $policy = undef,
+  Ferm::Tables $table = 'filter',
 ) {
+  # prevent unmanaged files due to new naming schema
+  # keep the default "filter" chains in the original location
+  # only prefix chains in other tables with the table name
+  if $table == 'filter' and $chain in ['INPUT', 'FORWARD', 'OUTPUT'] {
+    $filename = "${ferm::configdirectory}/chains/${chain}.conf"
+  } else {
+    $filename = "${ferm::configdirectory}/chains/${table}-${chain}.conf"
+  }
+
+  $builtin_chains = {
+    'raw'    => ['PREROUTING', 'OUTPUT'],
+    'nat'    => ['PREROUTING', 'INPUT', 'OUTPUT', 'POSTROUTING'],
+    'mangle' => ['PREROUTING', 'INPUT', 'FORWARD', 'OUTPUT', 'POSTROUTING'],
+    'filter' => ['INPUT', 'FORWARD', 'OUTPUT'],
+  }
+
+  if $policy and ! ($chain in $builtin_chains[$table]) {
+    fail('Can only set a default policy for builtin chains.')
```

Comment:
should we mention the policy/chain in the fail? this probably makes debugging easier if there are multiple chains with wrong policies

Reply:
good idea, done.